src/plugins/cal.js

```diff
@@ -0,0 +1,93 @@
+        "Note: Month value must be between 1-12 inclusive. Year can be any",
+        "4 digit number representing any Gregorian calendar value",
+        "",
+        "Examples",
+        "cal",
+        "cal 6",
+        "cal 4 2018"
+    ],
+    function(e) {
+        const monthNames = [
+            "Jan", "Feb", "Mar", "Apr", "May", "Jun",
+            "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+        ];
+        
+        const dayLine = "Sun Mon Tue Wed Thu Fri Sat";
+
+        let stringToParse = e.replace(/  +/g, ' ');
+        let parse = stringToParse.split(" ");
+        let warning = "";
+        try {
+            if (parse.length > 3) {
+                throw new Error("Too many arguments! Please see the function's usage with cal help")
+            } else if (parse.length > 1) {
+                if (isNaN(parse[1])) {
+                    throw new Error("Not a valid number for month!");
+                } else if (parse[1] < 1 || parse[1] > 12) {
+                    throw new Error("Invalid month requested");
+                }
+                    
+                if (parse.length == 2) {
+                    var year = new Date().getFullYear().toString();
+                    warning = "No valid year entered, assuming it is " + year + "\n";
+                } else if (isNaN(parse[2])) {
+                    throw new Error("Not a valid number for year!");
+                } else if (parse[2] < 0) {
+                    throw new Error("Invalid year requested - cannot be negative");
+                } else {
+                    var year = parseInt(parse[2]).toString();
+                }
+
+                var monthName = monthNames[parseInt(parse[1]) - 1];
+                var date = new Date (year, parseInt(parse[1]) - 1); 
+            } else {
+                var date = new Date();
+                var monthName = monthNames[date.getMonth()];
+                var year = date.getFullYear();
+                warning = "Printing for current month and year" + "\n";
+            }
+
+            (warning) ? this.parent.terminal.writeln(warning + "\n") : this.parent.terminal.writeln("\n");
+            
+            let monthLine = "         " + monthName + " " + year + "         ";
+            this.parent.terminal.writeln(monthLine);
+            this.parent.terminal.writeln(dayLine);
+            this.parent.terminal.writeln(generateCalString(date));
+        } catch(error) {
+            var val = "cal: " + error.message;
+            this.parent.terminal.writeln(val);
+        }
+    }
+};
+
+function generateCalString(date) {
+    let startOfMonth = new Date(date.getFullYear(), date.getMonth(), 1).getDay();
+    let numDays =  32 - new Date(date.getFullYear(), date.getMonth(), 32).getDate();
+    let daysIndex = startOfMonth + numDays;
+    var stringOut = "";
+
+    var i;
+    for (i = 0; i < daysIndex; i++) {
+        if (i >= startOfMonth) {
+            var date = i - startOfMonth + 1;
```

Comment:
Similar to above, it doesn't seem like you're benefitting from using `var` here.

Reply:
No real reason for var again, can change it